autoscaler/controllers/datacollection/daemonset.go

```diff
@@ -369,3 +379,29 @@ func patchDaemonSet(existing *appsv1.DaemonSet, desired *appsv1.DaemonSet, ctx c
 
 	return updated, nil
 }
+
+func isVersionSupported(cfg *rest.Config, minMajor int, minMinor int) (bool, error) {
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return false, err
+	}
+
+	serverVersion, err := discoveryClient.ServerVersion()
+	if err != nil {
+		return false, err
+	}
+
+	// Parse major and minor versions
+	major, err := strconv.Atoi(serverVersion.Major)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse major version: %w", err)
+	}
+
+	minor, err := strconv.Atoi(strings.TrimSuffix(serverVersion.Minor, "+"))
+	if err != nil {
+		return false, fmt.Errorf("failed to parse minor version: %w", err)
+	}
+
+	// Check if the server version meets or exceeds minMajor.minMinor
+	return major > minMajor || (major == minMajor && minor >= minMinor), nil
```

Comment:
This should be replaced by the standard functions of `apimachinery` for comparing versions